pkg/serverless/invocationlifecycle/lifecycle.go

```diff
@@ -280,32 +280,13 @@ func (lp *LifecycleProcessor) OnInvokeEnd(endDetails *InvocationEndDetails) {
 		spans = append(spans, span)
 
 		if lp.InferredSpansEnabled {
-			log.Debug("[lifecycle] Attempting to complete the inferred span")
-			log.Debugf("[lifecycle] Inferred span context: %+v", lp.GetInferredSpan().Span)
-			if lp.GetInferredSpan().Span.Start != 0 {
-				span0, span1 := lp.requestHandler.inferredSpans[0], lp.requestHandler.inferredSpans[1]
-				if span1 != nil {
-					log.Debug("[lifecycle] Completing a secondary inferred span")
-					lp.setParentIDForMultipleInferredSpans()
-					span1.AddTagToInferredSpan("http.status_code", statusCode)
-					span1.AddTagToInferredSpan("peer.service", lp.GetServiceName())
-					span := lp.completeInferredSpan(span1, lp.getInferredSpanStart(), endDetails.IsError)
-					spans = append(spans, span)
-					log.Debug("[lifecycle] The secondary inferred span attributes are %v", lp.requestHandler.inferredSpans[1])
-				}
-				span0.AddTagToInferredSpan("http.status_code", statusCode)
-				span0.AddTagToInferredSpan("peer.service", lp.GetServiceName())
-				span := lp.completeInferredSpan(span0, endDetails.EndTime, endDetails.IsError)
-				spans = append(spans, span)
-				log.Debugf("[lifecycle] The inferred span attributes are: %v", lp.GetInferredSpan())
-			} else {
-				log.Debug("[lifecyle] Failed to complete inferred span due to a missing start time. Please check that the event payload was received with the appropriate data")
-			}
+			inferredSpans := lp.endInferredSpan(statusCode, endDetails.EndTime, endDetails.IsError)
+			spans = append(spans, inferredSpans...)
 		}
 		lp.processTrace(spans)
 	}
 
-	if endDetails.IsError {
+	if endDetails.IsError && !endDetails.IsTimeout {
```

Comment:
Why exactly do we not want to create these metrics in the case of a timeout? Aren't timeouts the exact time when we'd want to create these metrics?

Reply:
We already handle the error metric in `serverless.go` for [timeouts](https://github.com/DataDog/datadog-agent/pull/21481/files/bf2379017eddefc45085430df45bbf3fb6d1fee5#diff-cfa9032f4e13abcf5afc327168c94e52ad3b382cd96ee6bad1615e9e29ae1864R141), so this was doubling error counts. I can leave a comment here so it's clearer

Reply:
Or would it make more sense to remove the metric from the other location and consolidate them here? Sounds like work for a follow up PR though.